client/dhcp_client_lwt.ml

```diff
@@ -0,0 +1,90 @@
+let src = Logs.Src.create "dhcp_client_lwt"
+module Log = (val Logs.src_log src : Logs.LOG)
+
+module Make(Time : Mirage_time_lwt.S) (Net : Mirage_net_lwt.S) = struct
+  open Lwt.Infix
+
+  type lease = Dhcp_wire.pkt
+
+  type t = lease Lwt_stream.t
+
+  let connect ?(with_xid)
+              ?(requests : Dhcp_wire.option_code list option) net =
```

Comment:
I don't think any sensible interface would include a `set_mac` function - at least the ARP implementation(s?!?) and other rely on the fact that the mac address of the underlying physical interface does not change.  For reading it from a database, it sounds like you'd like to have the `mac` function being inside the effectful `Lwt.t` layer.  This doesn't hurt IMHO, although if it can be solved another way (e.g. the initialisation function `connect` is effectful, why don't you read the mac address from the database in there and cache it?), it should be.